### PR TITLE
Fix issues with unicode var names

### PR DESCRIPTION
**Pull request by HirumalPriyashan**
## Purpose
- Fixes https://github.com/ballerina-platform/ballerina-lang/issues/36734

## Samples
image

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [x] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/utils/StringUtils.java [hunk 1]
```diff
@@ -172,4 +174,28 @@ public static String getErrorStringValue(Throwable error) {
...
+    /**
+     * Replace ballerina unicode format codes with their unicode character
+     * for a given string.
+     *
+     * @param toConvert String need to convert.
+     * @return Reformatted string.
+     */
+    public static String convertUnicodeToCharacter(String toConvert) {
+        Matcher matcher = Pattern.compile("\\\\u\\{([\\da-fA-F]*)}").matcher(toConvert);
+
```
Comment (DamithaSenevirathne) [inline ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/utils/StringUtils.java:187]: Shall we remove this newline ?

File: ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/utils/StringUtils.java [hunk 2]
```diff
@@ -172,4 +174,28 @@ public static String getErrorStringValue(Throwable error) {
...
+        StringBuilder stringBuilder = new StringBuilder();
+        int currentPosition = 0;
+        while (matcher.find()) {
+            stringBuilder.append(toConvert, currentPosition, matcher.start());
+            int code = Integer.parseInt(matcher.group(1), 16);
+            char[] chars = Character.toChars(code);
+            stringBuilder.append(chars);
```
Comment (DamithaSenevirathne) [inline ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/utils/StringUtils.java:194]: Shall we reduce this to a one line and avoid declaring `chars` if we are not using it later ? 
```suggestion
            stringBuilder.append(Character.toChars(code);
```

File: ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/utils/StringUtils.java (hunk 1 above)
Comment (HirumalPriyashan) [reply ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/utils/StringUtils.java]: Updated

File: ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/utils/StringUtils.java (hunk 2 above)
Comment (HirumalPriyashan) [reply ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/utils/StringUtils.java]: Updated

**Review by DamithaSenevirathne — DISMISSED** [review]
LGTM

**Review by dulajdilshan — APPROVED** [review]
LGTM

**Comment by DamithaSenevirathne** [comment]
This fix is already available in the master which was added by a sync PR related to 2201.2.x. branch.

**Pull request closed without merging**